vignettes/Resumable.Rmd

```diff
@@ -0,0 +1,203 @@
+
+---
+title: "Resumable Simulations"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Resumable}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE, message=FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  dpi=300,
+  fig.width = 7
+)
+```
+
+```{r setup, message=FALSE, class.source = 'fold-hide'}
+# Load the requisite packages:
+library(malariasimulation)
+library(dplyr)
+
+# Set colour palette:
+cols <- c("#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
+
+plot_incidence<- function(output, label){
+      output$incidence <-1000 *output$n_inc_clinical_0_1825 / output$n_0_1825
+      output$time_year<- output$timestep / 365
+      
+  plot(x = output$time_year, y = output$incidence, type = "l",
+     xlab = "Years", ylab = "Clinical incidence per 1,000 under 5", col = cols[1],
+     ylim = c(min(output$incidence)-0.25, max(output$incidence)+0.25),
+     xaxs = "i", yaxs = "i")
+    curve_values <- loess(incidence ~ time_year, data = output,
+                        span = 0.3, method = "loess") 
+  lines(output$time_year, predict(curve_values),
+        col = cols[5], lwd = 3)
+  title(main = label)
+}
+```
+
+In this vignette, we will demonstrate how to run a resumable malariasimulation model. This functionality can be useful to set time-varying parameters that can not be specified otherwise via malariasimulation helper functions. This can help save computational time in scenario modelling, where model parameters are the same between runs until an intervention is introduced.
+
+
+#   Run a simple simulation
+To begin, we can run a regular simulation using `malariasimulation::run_simulation` for 10 years.
+
+
+```{r}
+year <- 365
+month <- 30
+eir<- 35
+
+# pull standard parameters
+params <- get_parameters(
+  list(
+    human_population = 10000,
+    individual_mosquitoes = FALSE,
+    clinical_incidence_rendering_min_ages = 0,
+    clinical_incidence_rendering_max_ages = 5 * year
+  )
+)
+
+
+params <- set_equilibrium(parameters = params, init_EIR = eir)
+
+output<- run_simulation(params, timesteps= 10 * year)
+
+
+plot_incidence(output, label= 'Incidence for control run')
+
+
+
+```
+
+#   Run a resumable simulation
+
+Instead of running a simulation for the entire time horizon, we can choose to run an initial simulation, save the simulation state, and then resume at a specified point using `run_resumable_simulation`.
+
+The arguments are as follows:
+
+* *timesteps* : timestep to stop the simulation
+* *parameters*: input parameters
+* *correlations*: correlation parameters
+* *intial_state*: the state from which to resume the simulation (not needed for the first phase)
+* *restore_random_state*: boolean, choice to restore the random number generator's state from the checkpoint
```

Comment:
restore_random_state is an important feature based on conversations with people about stochasticity. Perhaps we should add a plot to the interventions example below where we show the implications of this parameter.

restore_random_state = FALSE is useful if you want to measure stochasticity of the outputs after the pause.
restore_random_state = TRUE is useful if you want a trajectory which would be consistent with an all-in-one simulation